lib/ansible/modules/cloud/ovirt/ovirt_vms.py

```diff
@@ -541,13 +574,52 @@ def __get_template_with_version(self):
 
         return template
 
+    def __get_storage_domain_and_all_template_disks(self, template):
+
+        if self._module.params.get('template') is None:
+            return None
+
+        if self.param('storage_domain') is None:
+            return None
+
+        storage_domain = self.param('storage_domain')
+
+        disk_attachments = (
+            self._connection.follow_link(template.disk_attachments)
+        )
+        disks = list()
+
+        for att in disk_attachments:
+            disks.append(
+                otypes.DiskAttachment(
+                    disk=otypes.Disk(
+                        id=att.disk.id,
+                        format=otypes.DiskFormat(self.param('disk_format')),
+                        storage_domains=[
+                            otypes.StorageDomain(
+                                id=get_id_by_name(
+                                    self._connection.system_service().storage_domains_service(),
+                                    storage_domain
+                                )
+                            )
+                        ]
+                    )
+                )
+            )
+
+        return disks
+
     def build_entity(self):
         template = self.__get_template_with_version()
+
+        disk_attachments = self.__get_storage_domain_and_all_template_disks(template)
+
         return otypes.Vm(
             name=self.param('name'),
             cluster=otypes.Cluster(
                 name=self.param('cluster')
             ) if self.param('cluster') else None,
+            disk_attachments=disk_attachments if disk_attachments else None,
```

Comment:
I think the `if` isn't needed here anymore as it's the job of `__get_storage_domain_and_all_template_disks` to handle it, no?